File: testdata/bench/javascript/BenchmarkTest00018.js
// Safe: SQL injection prevented via whitelist for role
const express = require('express');
const db = require('./db');
const router = express.Router();
const VALID_ROLES = ['admin', 'user', 'editor', 'viewer'];
router.get('/users/:role', (req, res) => {
    const role = req.params.role;
    if (!VALID_ROLES.includes(role)) return res.status(400).send('Invalid role');
    db.query("SELECT name, email FROM users WHERE role = ?", [role], (err, results) => {
        res.json(results);
    });
});
module.exports = router;

File: testdata/bench/c/BenchmarkTest00018.c
#include <sqlite3.h>

void get_admins(void) {
    sqlite3 *db;
    sqlite3_open("app.db", &db);
    sqlite3_exec(db, "SELECT * FROM users WHERE role = 'admin'", NULL, NULL, NULL);
    sqlite3_close(db);
}
